gtsfm/utils/verification.py

```diff
@@ -71,3 +56,32 @@ def fundamental_to_essential_matrix(
             Estimated essential matrix i2Ei1 as numpy array of shape (3x3).
     """
     return camera_intrinsics_i2.K().T @ i2Fi1 @ camera_intrinsics_i1.K()
+
+
+def compute_epipolar_distances(
+    normalized_coords_i1: np.ndarray, normalized_coords_i2: np.ndarray, i2Ei1: EssentialMatrix
+) -> np.ndarray:
+    """Compute symmetric point-line epipolar distances between normalized coordinates of correspondences.
+
+    Args:
+        normalized_coords_i1: normalized coordinates in image i1, of shape Nx2.
+        normalized_coords_i2: corr. normalized coordinates in image i2, of shape Nx2.
+        i2Ei1: essential matrix between two images
+
+    Returns:
+        Symmetric epipolar distances for each row of the input, of shape N.
+    """
+    # construct the essential matrix in the opposite directin
+    i2Ti1 = Pose3(i2Ei1.rotation(), i2Ei1.direction().point3())
+    i1Ti2 = i2Ti1.inverse()
+    i1Ei2 = EssentialMatrix(i1Ti2.rotation(), Unit3(i1Ti2.translation()))
```

Comment:
is this equivalent to `i1Ei2.T`?

Reply:
Yes it is. 

If there was a transpose function, we could use it directly. But the `EssentialMatrix` object internally stores a `Rot3` and `Unit3` so there is no direct way to transpose. I can move this exact logic to GTSAM if you think thats better.

Reply:
fine as it is, no worries